src/test/java/org/folio/circulation/domain/OverduePeriodCalculatorServiceTest.java

```diff
@@ -41,54 +39,25 @@
   private static final OverduePeriodCalculatorService calculator =
     new OverduePeriodCalculatorService(null, null);
 
-  @Test
-  public void preconditionsCheckLoanHasNoDueDate() {
-    DateTime systemTime = DateTime.now(UTC);
-    Loan loan = new LoanBuilder().asDomainObject();
-
-    assertFalse(calculator.preconditionsAreMet(loan, systemTime, true));
-  }
-
-  @Test
-  public void preconditionsCheckLoanDueDateIsInFuture() {
-    DateTime systemTime = DateTime.now(UTC);
-    Loan loan = new LoanBuilder()
-      .withDueDate(systemTime.plusDays(1))
-      .asDomainObject();
-
-    assertFalse(calculator.preconditionsAreMet(loan, systemTime, true));
-  }
-
-  @Test
-  public void preconditionsCheckCountClosedIsNull() {
-    DateTime systemTime = DateTime.now(UTC);
-    Loan loan = new LoanBuilder()
-      .withDueDate(systemTime.minusDays(1))
-      .asDomainObject();
-
-    assertFalse(calculator.preconditionsAreMet(loan, systemTime, null));
-  }
-
-  @Test
-  public void allPreconditionsAreMet() {
-    DateTime systemTime = DateTime.now(UTC);
-    Loan loan = new LoanBuilder()
-      .withDueDate(systemTime.minusDays(1))
-      .asDomainObject();
-
-    assertTrue(calculator.preconditionsAreMet(loan, systemTime, true));
-  }
-
   @Test
   public void countOverdueMinutesWithClosedDays() throws ExecutionException, InterruptedException {
     int expectedResult = MINUTES_PER_WEEK + MINUTES_PER_DAY + MINUTES_PER_HOUR + 1;
     DateTime systemTime = DateTime.now(UTC);
 
+    final OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(
+      new OverdueFinePolicyBuilder()
+        .withCountClosed(true)
+        .create());
+
     Loan loan = new LoanBuilder()
       .withDueDate(systemTime.minusMinutes(expectedResult))
-      .asDomainObject();
+      .asDomainObject()
+      .withOverdueFinePolicy(overdueFinePolicy);
+
+    LoanToChargeOverdueFine loanToChargeOverdueFine = new LoanToChargeOverdueFine(loan, null, policy -> true,
+      false, null, null, null, null, null);
```

Comment:
Yes, In a fact this method requires only a loan. But we need to wrap the loan into context object, to allow using aged to lost date instead of loan due date to calculate overdue minutes.